src/components/General/MessageBox.vue

```diff
@@ -173,13 +173,23 @@ export default {
           this.clearTargetRedirectToTargetListAndUpdateOperationStatus()
         }
       } else if (this.$isOnAgentView()) {
-        this.clearAgentEntitiesToDelete().then(response => {
-          if (response.status) {
-            this.updateOperationStatus(this.$entityStatus.SUCCESS, this.$message.successMessageForAgentDeletion)
-          } else {
-            this.updateOperationStatus(this.$entityStatus.FAILED, response.message)
-          }
-        })
+        if (this.entitiesToDelete.length > 1) {
```

Comment:
we need to refactor this method to many if else, I can not understand what we are doing there